tasks/smithy.rake

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+
+require 'rspec/core/rake_task'
+
+require_relative '../gems/smithy/spec/spec_helper'
+
+namespace :smithy do
+  RSpec::Core::RakeTask.new('spec:unit') do |t|
+    t.pattern = 'gems/smithy/spec/**/*_spec.rb'
+    t.rspec_opts = '--format documentation'
+    t.rspec_opts += ' --tag rbs_test' if ENV['SMITHY_RUBY_RBS_TEST']
+  end
+
+  task 'spec:endpoint_tests' do
+    generated_spec_task('endpoint_tests')
+  end
+
+  task 'spec:protocol_tests' do
+    generated_spec_task('protocol_tests')
+  end
+
+  task 'spec' => %w[spec:unit spec:endpoint_tests spec:protocol_tests]
```

Comment:
Could we move this to line 14? Just wanted to make the grouped-up rake task similar to other files. Same goes for rbs bulk task here